extensions/pkg/controller/healthcheck/actuator.go

```diff
@@ -58,13 +58,17 @@ type GetExtensionObjectListFunc = func() client.ObjectList
 // PreCheckFunc checks whether the health check shall be performed based on the given object and cluster.
 type PreCheckFunc = func(context.Context, client.Client, client.Object, *extensionscontroller.Cluster) bool
 
+// ErrorCodeCheckFunc checks if the given error is user specific error and return respective ErrorCodes.
+type ErrorCodeCheckFunc = func(string) []gardencorev1beta1.ErrorCode
+
 // ConditionTypeToHealthCheck registers a HealthCheck for the given ConditionType. If the PreCheckFunc is not nil it will
 // be executed with the given object before the health check if performed. Otherwise, the health check will always be
 // performed.
```

Comment:
Should we enhancce this docstring as well for the new function?

Reply:
I guess it is not necessary.